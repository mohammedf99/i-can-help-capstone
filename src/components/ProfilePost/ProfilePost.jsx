import React, { useContext, useEffect, useState } from "react";
import Link from "next/link";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashAlt } from "@fortawesome/free-solid-svg-icons";
import { PostCardCont, PostCardBody } from "./ProfilePost.styled";
import {
  deletePost,
  unPinPost,
  usersRef,
} from "../../Utilities/FirebaseUtilities";
import AuthContext from "../../Utilities/Contexts/AuthContext";

const ProfilePost = ({ data, isPin }) => {
  const [user, setUser] = useState(null);
  const postData = data?.post?.data();
  const { currentUser } = useContext(AuthContext);

  useEffect(() => {
    if (postData && postData.userId)
      usersRef
        .doc(postData?.userId)
        .onSnapshot((snapshot) => setUser(snapshot.data()));
  }, [postData]);

  const deleteHandler = (e) => {
    e.stopPropagation();
    if (!isPin) {
      deletePost(data?.post?.id, data?.post?.data()?.userId);
    } else {
      unPinPost(currentUser?.uid, data?.post?.id);
    }
  };

  return (
    <Link href={`/postDetails/${data?.post?.id}`}>
      <PostCardCont
        hoverable
        style={{ width: 240 }}
        cover={<img alt="example" src={postData?.picture} width="240" />}
      >
        <PostCardBody>
          <div className="upper-body">
            <h4 className="name">{user?.name}</h4>
            <h4 className="salary">{postData?.price}</h4>
          </div>
          <div className="middle-body">
            <p>{postData?.jobDescription}</p>
          </div>
          <div className="lower-body">
            <button onClick={deleteHandler} type="button">
              <FontAwesomeIcon icon={faTrashAlt} />
            </button>
          </div>
        </PostCardBody>
      </PostCardCont>
    </Link>
  );
};

export default ProfilePost;
