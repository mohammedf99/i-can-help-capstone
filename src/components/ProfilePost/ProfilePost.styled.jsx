import styled from "styled-components";
import { Card } from "antd";

export const PostCardCont = styled(Card)`
  border-radius: 10px;
  .ant-card-body {
    padding: 14px 16px;
  }
  a {
    color: #1c1259;
  }
  img {
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
  }
  p {
    margin-top: 0.75rem;
  }
  margin-right: 0.5rem;
  margin-bottom: 1rem;

  &:hover {
    box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  }
`;

export const PostCardBody = styled.div`
  display: flex;
  flex-direction: column;
  color: #1c1259;
  .upper-body {
    display: flex;
    justify-content: space-between;
  }
  .name,
  .salary {
    color: #1c1259;
    font-weight: 700;
    font-size: 1rem;
  }

  .middle-body {
    height: 70px;
    overflow: hidden;
    margin-bottom: 1rem;
  }
  a {
    text-decoration: none;
  }
  a:hover {
    color: #4e4e9f;
  }
  .lower-body {
    display: flex;
    justify-content: flex-end;
  }

  button {
    color: #ff9b00;
    background-color: transparent;
    border: none;
    outline: none;
    padding: 0.25rem 0.5rem;
    border-radius: 5px;
  }

  button:hover {
    background-color: #ff9b00;
    color: #ffffff;
  }
`;
