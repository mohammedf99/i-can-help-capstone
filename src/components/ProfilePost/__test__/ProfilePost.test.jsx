import * as React from "react";
import * as renderer from "react-test-renderer";
import ProfilePost from "../ProfilePost";

it("renders Navbar correctly", () => {
  const tree = renderer.create(<ProfilePost />).toJSON();
  expect(tree).toMatchSnapshot();
});
