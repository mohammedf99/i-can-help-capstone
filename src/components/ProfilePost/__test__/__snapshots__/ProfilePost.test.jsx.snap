// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable ProfilePoststyled__PostCardCont-sc-ab9iyo-0 kWaetX"
  style={
    Object {
      "width": 240,
    }
  }
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="example"
      src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png"
      width="240"
    />
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ProfilePoststyled__PostCardBody-sc-ab9iyo-1 jYerBU"
    >
      <div
        className="upper-body"
      >
        <h4
          className="name"
        >
          John Doe
        </h4>
        <h4
          className="salary"
        >
          $5/day
        </h4>
      </div>
      <div
        className="middle-body"
      >
        <a
          href="#"
        >
          <p>
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Amet tenetur dicta nobis ullam quisquam eius dolorem nesciunt minima error maxime natus, voluptate nihil delectus corporis fugiat id cupiditate iure in aliquid ut voluptatem non ex eligendi. Cum expedita dolorum accusantium beatae dolores laborum aliquid veritatis quod odit, consequuntur velit nesciunt.
          </p>
        </a>
      </div>
      <div
        className="lower-body"
      >
        <form
          action=""
        >
          <button
            type="submit"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-trash-alt fa-w-14 "
              data-icon="trash-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
